### PR TITLE
[ISSUE#120] update strategy and add offset_store test

### PR DESCRIPTION
**Pull request by luther-siyuan**
## What is the purpose of the change

- update AllocateByAveragely
- add AllocateByAveragely and test
- add AllocateByAveragelyCircle and test
- add AllocateByConfig and test
- add AllocateByMachineRoom and test
- add offset_store test
- add examples/consumer/strategy/main.go file

## Brief changelog

update strategy and add offset_store test

## Verifying this change

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when a cross-module dependency exists.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: examples/consumer/retry/order/main.go [hunk 1]
```diff
@@ -41,6 +41,7 @@ func main() {
 		consumer.WithConsumeFromWhere(consumer.ConsumeFromFirstOffset),
 		consumer.WithConsumerOrder(true),
 		consumer.WithMaxReconsumeTimes(5),
+		consumer.WithStrategy(consumer.AllocateByAveragely),
```
Comment (wenfengwang) [inline examples/consumer/retry/order/main.go]: please modify consumer/simple.go for this

File: consumer/strategy.go [hunk 2]
```diff
@@ -84,20 +84,46 @@ func AllocateByAveragely(consumerGroup, currentCID string, mqAll []*primitive.Me
...
+	)
+	for idx := range cidAll {
+		if cidAll[idx] == currentCID {
+			find = true
+			index = idx
+			break
+		}
+	}
+	if !find {
+		rlog.Infof("[BUG] ConsumerGroup=%s, ConsumerId=%s not in cidAll:%+v", consumerGroup, currentCID, cidAll)
```
Comment (wenfengwang) [inline consumer/strategy.go]: use warn level

Comment (luther-siyuan) [reply consumer/strategy.go]: ok

File: examples/consumer/retry/order/main.go (hunk 1 above)
Comment (luther-siyuan) [reply examples/consumer/retry/order/main.go]: have already added examples/consumer/strategy file

**Comment by wenfengwang** [comment]
LGTM

**Comment by jonnxu** [comment]
LGTM

**Pull request merged**